snippets/man_gen/hubl_custom_module_fields.json

```diff
@@ -0,0 +1,519 @@
+{
+  "Field Group":{
+    "prefix": "group.group",
+    "body": [
+      "{",
+        "\"id\": \"\",",
+        "\"name\": \"field_group\",",
+        "\"label\": \"Field group\",",
+        "\"sortable\": false,",
+        "\"required\": false,",
+        "\"locked\": false,",
+        "\"hierarchical\": false,",
+        "\"max_depth\": 0,",
+        "\"children\": [],",
+        "\"type\": \"group\",",
+        "\"default\": {}, }"              
+    ],
+    "description": "Field Group"
+  },
+  "Repeater":{
+    "prefix": "group.repeater",
+    "body":[
+      "{",
+      "  \"id\": \"\",",
+      "  \"name\": \"field_group\",",
+      "  \"label\": \"Field group\",",
+      "  \"sortable\": false,",
+      "  \"required\": false,",
+      "  \"locked\": false,",
+      "  \"occurrence\": {",
+      "    \"min\": null,",
+      "    \"max\": null,",
+      "    \"sorting_label_field\": null,",
+      "    \"default\": null",
+      "  },",
+      "  \"hierarchical\": false,",
+      "  \"max_depth\": 0,",
+      "  \"children\": [],",
+      "  \"type\": \"group\",",
+      "  \"default\": {} }"
+      
+    ],
+    "description": "Repeater Group"
+  },
+  "Date Field": {
+    "prefix": "field.date",
+    "body": [
+      "{",
+      "\"id\": \" \",",
+      "\"name\": \"${1:date_field}\",",
+      "\"label\": \"Date field\",",
+      "\"sortable\": false,",
+      "\"required\": false,",
+      "\"locked\": false,",
+      "\"hierarchical\": false,",
+      "\"max_depth\": 0,",
+      "\"step\": 1,",
+      "\"type\": \"date\",",
+      "\"default\": null }"
+    ],
+    "description": "Date field"
+  },
+  "Date Time Field": {
+    "prefix": "field.datetime",
+    "body": [
+      "{",
+      "\"id\": \"\",",
+      "\"name\": \"datetime_field\",",
+      "\"label\": \"Date and time field\",",
+      "\"sortable\": false,",
+      "\"required\": false,",
+      "\"locked\": false,",
+      "\"hierarchical\": false,",
+      "\"max_depth\": 0,",
+      "\"step\": 30,",
+      "\"type\": \"datetime\",",
+      "\"default\": null }"
+    ],
+    "description": "Date and Time Field"
+  },
+  "Link Field": {
+    "prefix": "field.link",
+    "body": [
+      "{",
+      "\"id\" : \"\",",
+      "\"name\" : \"link_field\",",
+      "\"label\" : \"Link field\",",
+      "\"sortable\" : false,",
+      "\"required\" : false,",
+      "\"locked\" : false,",
+      "\"hierarchical\" : false,",
+      "\"max_depth\" : 0,",
+      "\"supported_types\" : [ \"EXTERNAL\", \"CONTENT\", \"FILE\", \"EMAIL_ADDRESS\", \"BLOG\" ],",
+      "\"type\" : \"link\",",
+      "\"default\" : {",
+      "\"url\" : {",
+      "\"content_id\" : null,",
+      "\"type\" : \"EXTERNAL\",",
+      "\"href\" : \"\" },",
+      "\"open_in_new_tab\" : false,",
+      "\"no_follow\" : false",
+      "} }"
+    ],
+    "description": "Link Field"
```

Comment:
I wonder if all of the descriptions should be formatted as:
HubSpot Custom Module: Link Field
etc. with the "HubSpot Custom Module" prefix.

Thoughts?

Reply:
makes sense.  can amend.  This being my first time making a vscode snippet, i am unclear where the description is represented for the user.  If this is something that is visible, i would love to add some helpers here.  eg showing the tag parameters.  anything that helps the coder

Reply:
![field](https://user-images.githubusercontent.com/35824747/53097794-4b33eb80-34f0-11e9-9b73-edb2ac0f8698.gif)

Here is a gif of how it all plays out (my indentation is still messed up 😆 )! You can preview your local changes by clicking F5 and opening up a .json file. The description shows up in the intellisesne window! And you can see how the `${1|EXTERNAL,CONTENT,FILE,EMAIL_ADDRESS,BLOG|}` prompts the selection after initiating the snippet! 😄